### ISSUE TITLE
Use of TemporalEmbedding in non-Informer models

### ISSUE THREAD

**Issue by dylangrosz** [AIStream-Peelout/flow-forecast#337]
I am attempting to use temporal embeddings in the Simple Multi-Head Attention model but am running into issues with `temporal_decoding.py` with `simple_decode` in `decoder_params`. I have tried removing `decoder_params` but am running into dimension errors with `full_history` tensor. What is the best way to inject TemporalEmbeddings into non-Transformer models?

**Comment by isaacmg** [comment]
Honestly surprised that is working at all.  The only model at the moment that support using TemporalEmbeddings is the `Informer` model which takes it as a separate parameter. Could you describe in more detail what you are trying to achieve?

**Comment by dylangrosz** [comment]
I am trying to take the temporal embedding concept and apply it to the Simple Multi-Head Attention model. I've also tried directly editing the Simple MHA model by removing the first dense layer and adding a temporal embedding, but I get hung up in evaluation at `generate_predictions_non_decoded`.

**Comment by isaacmg** [comment]
Yeah you probably want to use the decoded predictions, specifically https://github.com/AIStream-Peelout/flow-forecast/blob/5ebd89dcb758ea5d3fea9e3bd0ece94566ba3026/flood_forecast/temporal_decoding.py however that is Informer specific. The `generate_predictions_non_decoded` hasn't been tested in a while and might be dropped entirely in the future. Th

**Comment by dylangrosz** [comment]
A generalized temporal embedding + temporal decoding function would be a great enhancement! This repo is doing great work centralizing many SotA time series models, and it would be helpful for experimentation to allow the transfer of different models' modules to enhance customization. For now, I'll try to hack a solution to get temporal embeddings to work with the MHA model, but I'll keep watching this repo for more enhancements.
